File: public/docs/_examples/dependency-injection/ts/app/heroes/heroes.component.1.ts
// #docplaster
// #docregion full, v1
import { Component }          from '@angular/core';

// #enddocregion full, v1
import { HeroListComponent }  from './hero-list.component.2';
import { HeroService }        from './hero.service.1';
/*
// #docregion full
import { HeroListComponent }  from './hero-list.component';
// #docregion v1
import { HeroService }        from './hero.service';
// #enddocregion full, v1
*/
// #docregion full, v1

@Component({
  selector: 'my-heroes',
  template: `
  <h2>Heroes</h2>
  <hero-list></hero-list>
  `,
  // #enddocregion v1
  providers: [HeroService],
  // #docregion v1
  directives: [HeroListComponent]
})
export class HeroesComponent { }

File: public/docs/_examples/dependency-injection/dart/lib/heroes/heroes_component_1.dart
// #docplaster
// #docregion
// #docregion full, v1
import 'package:angular2/core.dart';

// #enddocregion full, v1
import 'hero_list_component_2.dart';
import 'hero_service_1.dart';
/*
// #docregion full, v1
import 'hero_list_component.dart';
// #enddocregion v1
import 'hero_service.dart';
// #enddocregion full
*/
// #docregion full, v1

@Component(
    selector: 'my-heroes',
    template: '''
      <h2>Heroes</h2>
      <hero-list></hero-list>''',
    // #enddocregion v1
    providers: const [HeroService],
    // #docregion v1
    directives: const [HeroListComponent])
class HeroesComponent {}
